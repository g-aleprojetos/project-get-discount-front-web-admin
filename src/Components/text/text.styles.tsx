import React from 'react';
import styled from 'styled-components';
import cores from 'resources/colors';

export type Props = {
  id?: string;
  cor?: string;
  fontSize?: string;
  marginTop?: number;
  children?: string | React.ReactNode;
  peso?: 'bold' | 'medium';
  toUpper?: boolean;
  cursor?: 'default' | 'pointer' | 'text';
};

const TextoBase = styled.p<Props>`
  font-size: ${props => props.fontSize ?? '12'}px;
  color: ${props => props.cor ?? cores.white};
  margin-top: ${props => props.marginTop ?? 0}px;
  text-transform: ${props => (props.toUpper ? 'uppercase' : 'none')};
  cursor: ${props => props.cursor ?? 'text'};
`;

export const Texto = {
  P: TextoBase,
  H1: TextoBase.withComponent('h1'),
  H2: TextoBase.withComponent('h2'),
  H3: TextoBase.withComponent('h3'),
  H4: TextoBase.withComponent('h4'),
  H5: TextoBase.withComponent('h5'),
  H6: TextoBase.withComponent('h6'),
};
