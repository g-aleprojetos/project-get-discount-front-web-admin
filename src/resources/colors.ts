export default {
  caribbeanGreen: '#04c582',
  mineShaft: '#2e3137',
  shuttleGray: '#545d6a',
  lightgray: '#d3d3d3',
  white: '#ffffff',
  red: '#ff0000',
  blue: '#3D9DB3',
  black: '#000000',
};
