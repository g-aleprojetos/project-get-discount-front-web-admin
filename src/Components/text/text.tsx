import React from 'react';

import {Texto, Props} from './text.styles';
import texts from 'resources/texts';

export function TextH1(props: Props) {
  return (
    <Texto.H1
      tamanho={texts.tamanho.medium}
      alturaDeLinha={texts.altura.medium}
      {...props}
    />
  );
}

export function TextH2(props: Props) {
  return (
    <Texto.H2
      tamanho={texts.tamanho.small}
      alturaDeLinha={texts.altura.small}
      {...props}
    />
  );
}

export function TextH3(props: Props) {
  return (
    <Texto.H3
      tamanho={texts.tamanho.xsmall}
      alturaDeLinha={texts.altura.xsmall}
      {...props}
    />
  );
}

export function TextH4(props: Props) {
  return (
    <Texto.H4
      tamanho={texts.tamanho.xxsmall}
      alturaDeLinha={texts.altura.xxsmall}
      {...props}
    />
  );
}

export function TextH5(props: Props) {
  return (
    <Texto.H5
      tamanho={texts.tamanho.xxxsmall}
      alturaDeLinha={texts.altura.xxxsmall}
      {...props}
    />
  );
}

export function TextH6(props: Props) {
  return (
    <Texto.H6
      tamanho={texts.tamanho.xxxxsmall}
      alturaDeLinha={texts.altura.xxxxsmall}
      {...props}
    />
  );
}

export function TextP(props: Props) {
  return (
    <Texto.P
      tamanho={texts.tamanho.xsmall}
      alturaDeLinha={texts.altura.xsmall}
      {...props}
    />
  );
}
