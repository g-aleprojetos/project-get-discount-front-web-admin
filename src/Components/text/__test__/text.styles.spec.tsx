import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import {Texto} from '../text.styles';
import colors from 'resources/colors';

describe('Text.styles', () => {
  test('Texto "H1" DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<Texto.H1 />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "H2" DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<Texto.H2 />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "H3" DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<Texto.H3 />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "H4" DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<Texto.H4 />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "H5" DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<Texto.H5 />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "H6" DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<Texto.H6 />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "P" DEVE ser igual ao snapshot', () => {
    const container = renderer.create(<Texto.P />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test(`Texto "P" DEVE ter a cor ${colors.red} QUANDO for passada pelo props`, () => {
    const container = renderer.create(<Texto.P cor={colors.red} />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "P" DEVE estar com o texto em "uppercase" QUANDO for passado pelo props', () => {
    const container = renderer.create(<Texto.P toUpper={true} />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "P" DEVE ter o cursor como "pointer" QUANDO for passado "pointer" no cursor da props', () => {
    const container = renderer.create(<Texto.P cursor="pointer" />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "P" DEVE ter o cursor como "default" QUANDO for passado "default" no cursor da props', () => {
    const container = renderer.create(<Texto.P cursor="default" />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "P" DEVE ter o marginTop como "32" QUANDO for passado "32" no marginTop da props', () => {
    const container = renderer.create(<Texto.P marginTop={32} />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "P" DEVE ter o peso como "bold" QUANDO for passado "bold" no peso da props', () => {
    const container = renderer.create(<Texto.P peso="bold" />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "P" DEVE ter o peso como "medium" QUANDO for passado "medium" no peso da props', () => {
    const container = renderer.create(<Texto.P peso="medium" />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "P" DEVE ter o tamanho como "32" QUANDO for passado "32" no tamanho da props', () => {
    const container = renderer.create(<Texto.P tamanho={32} />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "P" DEVE ter o alturaDeLinha como "32" QUANDO for passado "32" no alturaDeLinha da props', () => {
    const container = renderer.create(<Texto.P alturaDeLinha={32} />).toJSON();

    expect(container).toMatchSnapshot();
  });

  test('Texto "P" DEVE ter o fontSize como "32" QUANDO for passado "32" no fontSize da props', () => {
    const container = renderer.create(<Texto.P fontSize="32px" />).toJSON();

    expect(container).toMatchSnapshot();
  });
});
