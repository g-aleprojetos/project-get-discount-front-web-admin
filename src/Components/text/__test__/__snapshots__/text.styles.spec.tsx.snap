// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text.styles Texto "H1" DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: px;
  cursor: text;
}

<h1
  className="c0"
/>
`;

exports[`Text.styles Texto "H2" DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: px;
  cursor: text;
}

<h2
  className="c0"
/>
`;

exports[`Text.styles Texto "H3" DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: px;
  cursor: text;
}

<h3
  className="c0"
/>
`;

exports[`Text.styles Texto "H4" DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: px;
  cursor: text;
}

<h4
  className="c0"
/>
`;

exports[`Text.styles Texto "H5" DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: px;
  cursor: text;
}

<h5
  className="c0"
/>
`;

exports[`Text.styles Texto "H6" DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: px;
  cursor: text;
}

<h6
  className="c0"
/>
`;

exports[`Text.styles Texto "P" DEVE estar com o texto em "uppercase" QUANDO for passado pelo props 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: uppercase;
  line-height: px;
  cursor: text;
}

<p
  className="c0"
/>
`;

exports[`Text.styles Texto "P" DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: px;
  cursor: text;
}

<p
  className="c0"
/>
`;

exports[`Text.styles Texto "P" DEVE ter a cor #f4715D QUANDO for passada pelo props 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #f4715D;
  margin-top: 0px;
  text-transform: none;
  line-height: px;
  cursor: text;
}

<p
  className="c0"
/>
`;

exports[`Text.styles Texto "P" DEVE ter o alturaDeLinha como "32" QUANDO for passado "32" no alturaDeLinha da props 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: 32px;
  cursor: text;
}

<p
  className="c0"
/>
`;

exports[`Text.styles Texto "P" DEVE ter o cursor como "default" QUANDO for passado "default" no cursor da props 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: px;
  cursor: default;
}

<p
  className="c0"
  cursor="default"
/>
`;

exports[`Text.styles Texto "P" DEVE ter o cursor como "pointer" QUANDO for passado "pointer" no cursor da props 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: px;
  cursor: pointer;
}

<p
  className="c0"
  cursor="pointer"
/>
`;

exports[`Text.styles Texto "P" DEVE ter o fontSize como "32" QUANDO for passado "32" no fontSize da props 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: px;
  cursor: text;
}

<p
  className="c0"
  fontSize="32px"
/>
`;

exports[`Text.styles Texto "P" DEVE ter o marginTop como "32" QUANDO for passado "32" no marginTop da props 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 32px;
  text-transform: none;
  line-height: px;
  cursor: text;
}

<p
  className="c0"
/>
`;

exports[`Text.styles Texto "P" DEVE ter o peso como "bold" QUANDO for passado "bold" no peso da props 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Bold;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: px;
  cursor: text;
}

<p
  className="c0"
/>
`;

exports[`Text.styles Texto "P" DEVE ter o peso como "medium" QUANDO for passado "medium" no peso da props 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: px;
  cursor: text;
}

<p
  className="c0"
/>
`;

exports[`Text.styles Texto "P" DEVE ter o tamanho como "32" QUANDO for passado "32" no tamanho da props 1`] = `
.c0 {
  font-size: 32px;
  font-family: AvertaStd-Regular;
  color: #000000;
  margin-top: 0px;
  text-transform: none;
  line-height: px;
  cursor: text;
}

<p
  className="c0"
/>
`;
